modules/phase_field/src/action/PolycrystalVoronoiCoupledVoidICAction.C

```diff
@@ -0,0 +1,64 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "PolycrystalVoronoiCoupledVoidICAction.h"
+#include "PolycrystalVoronoiCoupledVoidIC.h"
+#include "Factory.h"
+#include "FEProblem.h"
+#include "Conversion.h"
+
+registerMooseAction("PhaseFieldApp", PolycrystalVoronoiCoupledVoidICAction, "add_ic");
+
+InputParameters
+PolycrystalVoronoiCoupledVoidICAction::validParams()
+{
+  InputParameters params = Action::validParams();
+  params += PolycrystalVoronoiCoupledVoidIC::actionParameters();
+  params.addClassDescription("Automatically creates initial conditions for each grain"
+                             "order parameter of the polycrystal voronoi coupled with voids");
+  params.addRequiredParam<std::string>("var_name_base", "specifies the base name of the variables");
+  params.suppressParameter<VariableName>("variable");
+  params.addRequiredParam<UserObjectName>(
+      "polycrystal_ic_uo", "UserObject for obtaining the polycrystal grain structure.");
+  params.addParam<FileName>(
+      "file_name",
+      "",
+      "File containing grain centroids, if file_name is provided, the centroids "
+      "from the file will be used.");
+  return params;
+}
```

Comment:
This is pretty much a copy of `PolycrystalVoronoiVoidICAction`. I'd just derive this from that class and override the act method. I also don't see you coupling in the void order parameter.

Reply:
actually, I see it now